crates/ethereum/evm/src/execute.rs

```diff
@@ -209,13 +227,17 @@ where
                 gas: GotExpected { got: cumulative_gas_used, expected: block.gas_used },
                 gas_spent_by_tx: receipts.gas_spent_by_tx()?,
             }
-            .into())
+            .into());
         }
 
+        // Collect all EIP-6110 deposits
+        let deposits = parse_deposits_from_receipts(&receipts)?;
+
         // Collect all EIP-7685 requests
         let withdrawal_requests =
             post_block_withdrawal_requests(&self.chain_spec, block.timestamp, &mut evm)?;
```

Comment:
Should accumulate, extending a vector, rather than two vecs and concatenating after

Reply:
The withdrawal function returns a `Vec<Request>` via `Vec<Request>::decode` and I cannot break out of the iterator without collecting so not sure how to do that here, also prob not big perf given smol list

Reply:
i see, what about moving the parse_deposits to after post_block and extending that list?